corehq/ex-submodules/casexml/apps/phone/models.py

```diff
@@ -700,10 +700,11 @@ def get_cases_that_directly_depend_on_case(self, case_id):
         return self.reverse_indices.get(case_id, [])
 
     def delete_index(self, from_case_id, index_name):
-        prior_ids = self.indices.pop(from_case_id, {})
-        prior_ids.pop(index_name, None)
-        if prior_ids:
-            self.indices[from_case_id] = prior_ids
+        case_indices = self.indices.get(from_case_id, {})
+        if len(case_indices) > 1:
+            case_indices.pop(index_name, {})
+        elif len(case_indices) == 1:
+            self.indices.pop(from_case_id)
```

Comment:
This was an attempt to fix this:
```    
89240    0.617    0.000   31.671    0.000 models.py:708(indices)
```
but I don't think it worked, as that profile line didn't decrease when I made this change. 


the cumtime of something in here is 31.671s, which is nuts. It also looks like a large proportion of the time is taken up with `JsonObject` stuff, which I think is going to be hard to optimize around (other than reducing dict lookups).